R/descriptives.R

```diff
@@ -0,0 +1,154 @@
+#'   of variation. If supplied, `x` is not used to compute the mean.
+#' @param sigma A numeric vector of standard deviation values to use to compute the coefficient
+#'   of variation. If supplied, `x` is not used to compute the SD.
+#' @param method Method to use to compute the CV. Can be `"standard"` to compute
+#'   by dividing the standard deviation by the mean, `"unbiased"` for the
+#'   unbiased estimator for normally distributed data, or one of two robust
+#'   alternatives: `"median_mad"` to divide the median by the [stats::mad()],
+#'   or `"qcd"` (quartile coefficient of dispersion, interquartile range divided
+#'   by the sum of the quartiles \[twice the midhinge\]: \eqn{(Q_3 - Q_1)/(Q_3 + Q_1)}.
+#' @param trim the fraction (0 to 0.5) of values to be trimmed from
+#'   each end of `x` before the mean and standard deviation (or alternatvies)
+#'   are computed. Values of `trim` outside the range of (0 to 0.5) are taken
+#'   as the nearest endpoint.
+#' @param na.rm Logical. Should `NA` values be removed before computing (`TRUE`)
+#'   or not (`FALSE`, default)?
+#' @param n If `method = "unbiased"` and both `mu` and `sigma` are provided (not
+#'   computed from `x`), what sample size to use to adjust the computed CV
+#'   for small-sample bias?
+#' @param ... Further arguments passed to computation functions.
+#'
+#' @rdname coef_var
+#'
+#' @export
+coef_var.numeric <- function(x, mu = NULL, sigma = NULL,
+                             method = c("standard", "unbiased", "median_mad", "qcd"),
+                             trim = 0, na.rm = FALSE, n = NULL, ...) {
+  # TODO: Support weights
+  method <- match.arg(method, choices = c("standard", "unbiased", "median_mad", "qcd"))
+  if (is.null(mu) || is.null(sigma)) {
+    if (isTRUE(na.rm)) {
+      x <- .drop_na(x)
+    }
+    n <- length(x)
+    x <- .trim_values(x, trim = trim, n = n)
+  }
+  if (is.null(mu)) {
+    mu <- switch(
+      method,
+      standard = , unbiased = mean(x, ...),
+      median_mad = stats::median(x, ...),
+      qcd = unname(sum(stats::quantile(x, probs = c(.25, .75), ...)))
+    )
+  }
+  if (is.null(sigma)) {
+    sigma <- switch(
+      method,
+      standard = , unbiased = stats::sd(x, ...),
+      median_mad = stats::mad(x, center = mu, ...),
+      qcd = unname(diff(stats::quantile(x, probs = c(.25, .75), ...)))
+    )
+  }
+  out <- sigma / mu
+  if (method == "unbiased") {
+    if (is.null(n)) {
+      stop(insight::format_message(
+        "A value for `n` must be provided when `method = \"unbiased\"` and both `mu` and `sigma` are provided."
+      ), call. = FALSE)
+    }
+    # from DescTools::CoefVar
+    out <- out * (1 - 1 / (4 * (n - 1)) + 1 / n * out^2 + 1 / (2 * (n - 1)^2))
+  }
+  return(out)
+}
+
+
+
+
+# descriptives helpers
+
+.drop_na <- function(x) {
+  x[!is.na(x)]
+}
+
+.trim_values <- function(x, trim = 0, n = NULL, weights = NULL) {
+  # TODO: Support weights
```

Comment:
@bwiernik Can this already be merged, or do you also want to work in this PR on supporting weights here?

Reply:
We should at least print a message if `weights` has no functionality yet.